### ISSUE TITLE
PayrollAU dates issue

### ISSUE THREAD

**Issue by pauliak1** [XeroAPI/xero-php-oauth2#106]
What is the format and timezone used for storing StartDate and DateOfBirth of Employee Object in PayrollAU?

If I use the setStartDateAsDate/setDateOfBirthAsDate function when creating an employee it keeps setting the date 1 day prior to the date I am setting.

e.g.
$employee = new XeroAPI\XeroPHP\Models\PayrollAu\Employee;
$dateOfBirth = DateTime::createFromFormat('Y-m-d', '1980-10-25');
$employee->setDateOfBirthAsDate($dateOfBirth);

it sets the date of birth as 24 Oct 1980 (instead of 25 Oct 1980)

**Comment by SidneyAllen** [comment]
@pauliak1 - thanks for the feedback.  There is a conversion from date to seconds to get dates into MS Date format.  If you have time to play around with this and make a a PR, the method is located here https://github.com/XeroAPI/xero-php-oauth2/blob/d6f2c140bd2c48d611b6a051e6af5cccc7537294/lib/StringUtil.php#L57

**Comment by SidneyAllen** [comment]
@pauliak1 - thanks for the feedback.  There is a conversion from date to seconds to get dates into MS Date format. 

https://github.com/XeroAPI/xero-php-oauth2/blob/d6f2c140bd2c48d611b6a051e6af5cccc7537294/lib/Models/PayrollAu/Employee.php#L513

We can't accept a direct PR, but If you have time to play around with this and suggest a fix, we can update our templates used to generate the models.

**Comment by pauliak1** [comment]
The issue is not the code but the old date format used (e.g. /Date(1198908717056+0000)/ ). It was used prior to JSON.NET 4.5 which was over 8 years ago. 
Basically we need to know what timezone the server is running on, which I assume is UTC.

To overcome the issue we add 10 hours to the date. (Australian Easter Time is UTC+10) 

$timeInMillis = strtotime($dateOfBirth." 10:00:00") * 1000;
$dateOfBirth = "/Date(" . $timeInMillis. "+0000)/";
$employee->setDateOfBirth($dateOfBirth);

Not the cleanest solution but works for now.

**Comment by SidneyAllen** [comment]
@pauliak1 Thank you for your feedback.

I've updated our templates to add the UTC param when performing the "strtotime" conversion.

I did a test last night and created an employee (I'm in PST), so 7 hour difference.  I found the dateOfBirth I submitted was the same when looking in Xero UI.  So, fairly confident this resolves the issue with AU Payroll and date conversion.

Give it a try and let us know.
